platform/cog-platform-fdo.c

```diff
@@ -1306,9 +1556,19 @@ clear_wayland (void)
     if (wl_data.shell != NULL)
         wl_shell_destroy (wl_data.shell);
 
+    if (wl_data.subcompositor != NULL)
+        wl_subcompositor_destroy (wl_data.subcompositor);
```

Comment:
These two lines could just be `g_clear_pointer (&wl_data.subcompositor, wl_subcompositor_destroy);`.